src/lib.rs

```diff
@@ -3799,46 +3795,96 @@ safety_comment! {
-);
+#[cfg(zerocopy_atomics)]
+#[cfg(target_has_atomic = "8")]
+#[cfg_attr(doc_cfg, doc(cfg(target_has_atomic = "8")))]
+mod atomic_8 {
+    use super::*;
+    unsafe_impl_traits_for_atomics!(AtomicBool[bool], AtomicU8[u8], AtomicI8[i8]);
+    safety_comment! {
+        /// SAFETY:
+        /// Per [1], `AtomicBool`, `AtomicU8`, and `AtomicI8` have an alignment of 1.
+        ///
+        /// [1] Per https://doc.rust-lang.org/nightly/core/sync/atomic/struct.AtomicBool.html:
+        ///
+        ///     This type has the same size, alignment, and bit validity as [the native type].
+        unsafe_impl!(AtomicBool: Unaligned);
+        unsafe_impl!(AtomicU8: Unaligned);
+        unsafe_impl!(AtomicI8: Unaligned);
+    }
+    assert_unaligned!(AtomicBool, AtomicU8, AtomicI8);
+}
+#[cfg(zerocopy_atomics)]
+#[cfg(target_has_atomic = "16")]
+#[cfg_attr(doc_cfg, doc(cfg(target_has_atomic = "16")))]
+mod atomic_16 {
+    use super::*;
+    unsafe_impl_traits_for_atomics!(AtomicU16[u16], AtomicI16[i16]);
+}
+#[cfg(zerocopy_atomics)]
+#[cfg(target_has_atomic = "32")]
+#[cfg_attr(doc_cfg, doc(cfg(target_has_atomic = "32")))]
+mod atomic_32 {
+    use super::*;
+    unsafe_impl_traits_for_atomics!(AtomicU32[u32], AtomicI32[i32]);
+}
+#[cfg(zerocopy_atomics)]
+#[cfg(target_has_atomic = "64")]
+#[cfg_attr(doc_cfg, doc(cfg(target_has_atomic = "64")))]
+mod atomic_64 {
+    use super::*;
+    unsafe_impl_traits_for_atomics!(AtomicU64[u64], AtomicI64[i64]);
+}
+#[cfg(zerocopy_atomics)]
+#[cfg(target_has_atomic = "ptr")]
+#[cfg_attr(doc_cfg, doc(cfg(target_has_atomic = "ptr")))]
+mod atomic_ptr {
+    use super::*;
+    unsafe_impl_traits_for_atomics!(AtomicUsize[usize], AtomicIsize[isize]);
 
-safety_comment! {
     /// SAFETY:
-    /// Per [1], `AtomicBool`, `AtomicU8`, and `AtomicI8` have the same size as
-    /// `bool`, `u8`, and `i8` respectively. Since a type's alignment cannot be
-    /// smaller than 1 [2], and since its alignment cannot be greater than its
-    /// size [3], the only possible value for the alignment is 1. Thus, it is
-    /// sound to implement `Unaligned`.
+    /// `AtomicPtr<T>`` is garunteed to wrap a `*mut T`. Just like *mut T, we
+    /// don't implement FromBytes/IntoBytes.
     ///
-    /// [1] TODO(#896), TODO(https://github.com/rust-lang/rust/pull/121943):
-    ///     Cite docs once they've landed.
-    ///
-    /// [2] Per https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment:
-    ///
-    ///     Alignment is measured in bytes, and must be at least 1.
-    ///
-    /// [3] Per https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment:
-    ///
-    ///     The size of a value is always a multiple of its alignment.
-    unsafe_impl!(AtomicBool: Unaligned);
-    unsafe_impl!(AtomicU8: Unaligned);
-    unsafe_impl!(AtomicI8: Unaligned);
-    assert_unaligned!(AtomicBool, AtomicU8, AtomicI8);
+    /// See the comments on the `*mut T` impls for more information.
```

Comment:
This safety comment smells to me. We shouldn't have to talk about `FromBytes`/`IntoBytes` in order to justify this impl - we should only need to reference the safety invariants on `util::Atomic`.